diff --git a/node_modules/hast-util-to-html/lib/element.js b/node_modules/hast-util-to-html/lib/element.js
index 7ad832e..66ea6b5 100644
--- a/node_modules/hast-util-to-html/lib/element.js
+++ b/node_modules/hast-util-to-html/lib/element.js
@@ -28,6 +28,11 @@ function serializeElement(ctx, node, index, parent) {
     ctx.schema = svg
   }
 
+  // Hackfix <a> tags to have target="_blank"
+  if (node.tagName === "a") {
+	node.properties.target = "_blank";
+  }
+
   attrs = serializeAttributes(ctx, node.properties)
 
   content = all(
